arkouda/numeric.py

```diff
@@ -2001,3 +2027,76 @@ def value_counts(
     (array([0, 2, 4]), array([3, 2, 1]))
     """
     return GroupBy(pda).count()
+
+
+@typechecked
+def clip(
+    pda: pdarray,
+    lo: Union[numeric_scalars, pdarray],
+    hi: Union[numeric_scalars, pdarray],
+) -> pdarray:
+    """
+    Mimic the behavior of numpy.clip.
```

Comment:
so for the description, we prefer for it to be self contained. We want a user who isn't familiar with `np.clip` to have good idea of what this function will do after reading it. You can refer to the docstrings of other arkouda functions for examples

I like to use the description of the function I'm mimicking as a jumping off point. The description of [`np.clip`](https://numpy.org/doc/stable/reference/generated/numpy.clip.html) is:
> Clip (limit) the values in an array.
> 
> Given an interval, values outside the interval are clipped to the interval edges. For example, if an interval of [0, 1] is specified, values smaller than 0 become 0, and values larger than 1 become 1.